components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -4613,13 +4615,16 @@ public static ApiEndpointValidationResponseDTO sendHttpHEADRequest(String urlVal
      */
     @Override
     public Response getAPIRevisions(String apiId, MessageContext messageContext) {
-        // remove errorObject and add implementation code!
-        ErrorDTO errorObject = new ErrorDTO();
-        Response.Status status = Response.Status.NOT_IMPLEMENTED;
-        errorObject.setCode((long) status.getStatusCode());
-        errorObject.setMessage(status.toString());
-        errorObject.setDescription("The requested resource has not been implemented");
-        return Response.status(status).entity(errorObject).build();
+        try {
+            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+            List<APIRevision> apiRevisions = apiProvider.getAPIRevisions(apiId);
+            APIRevisionListDTO apiRevisionListDTO = APIMappingUtil.fromListAPIRevisiontoDTO(apiRevisions);
+            return Response.ok().entity(apiRevisionListDTO).build();
+        } catch (Exception e) {
```

Comment:
Why catching a generic Exception

Reply:
changed to APIManagementException, resolved